src/dragen_align_pa/stages.py

```
@@ -11,100 +12,225 @@
 
         ica_prep_job: PythonJob = prepare_ica_for_analysis.run_ica_prep_job(
-            sequencing_group=sequencing_group,
-            output=str(outputs),
-            api_root=ICA_REST_ENDPOINT,
-            sg_name=sequencing_group.name,
-            bucket_name=bucket_name,
+            cohort=cohort,
+            output=outputs,
         )
 
         return self.make_outputs(
-            target=sequencing_group,
-            data=outputs,
+            target=cohort,
+            data=outputs,  # pyright: ignore[reportArgumentType]
             jobs=ica_prep_job,
         )
 
 
+@stage(required_stages=[PrepareIcaForDragenAnalysis])
+class FastqIntakeQc(CohortStage):
+    """Generate md5 sums for each uploaded fastq file.
+
+    Check these sums against the supplied md5sums to check for any corruption in transit.
+    """
+
+    def expected_outputs(self, cohort: Cohort) -> dict[str, cpg_utils.Path]:  # pyright: ignore[reportIncompatibleMethodOverride]
+        intake_qc_results: dict[str, cpg_utils.Path] = {
+            'fastq_ids_outpath': get_prep_path(filename=f'{cohort.name}_fastq_ids.txt'),
+            'md5sum_pipeline_run': get_prep_path(filename=f'{cohort.name}_ica_md5sum_pipeline.json'),
+            'md5sum_pipeline_success': get_prep_path(filename=f'{cohort.name}_md5_pipeline_success'),
+            f'{cohort.name}_md5_errors': get_prep_path(filename=f'{cohort.name}_md5_errors.log'),
+        }
+        return intake_qc_results
+
+    def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:  # noqa: ARG002
+        if READS_TYPE == 'fastq':
+            outputs: dict[str, cpg_utils.Path] = self.expected_outputs(cohort=cohort)
+            md5job: PythonJob = manage_md5_pipeline.run_md5_management_job(cohort=cohort, outputs=outputs)
+
+            return self.make_outputs(target=cohort, data=outputs, jobs=md5job)  # pyright: ignore[reportArgumentType]
+
+        return None
```

I think we should raise an error here that tells the user they forgot to set the config `['workflow', 'reads_type']` as fastq instead of the workflow likely just saying 'no jobs to submit'. If raising an error it will get printed to the batch screen and can be quickly fixed.

---

I have added this check into `constants.py`